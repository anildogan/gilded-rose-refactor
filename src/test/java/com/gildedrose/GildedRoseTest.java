package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class GildedRoseTest {

    @Test
    void foo() {
        CombinationApprovals.verifyAllCombinations(this::doGetItem,
            new String[]{"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"},
            new Integer[]{0, 9, 10, 11, -1},
            new Integer[]{0,1,2,49,51});
    }

    private String doGetItem(String name, int sellIn, int quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }

}
