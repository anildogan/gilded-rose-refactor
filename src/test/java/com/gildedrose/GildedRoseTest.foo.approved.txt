[foo, 0, 0] => foo, -1, 0 
[foo, 0, 1] => foo, -1, 0 
[foo, 0, 2] => foo, -1, 0 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 51] => foo, -1, 49 
[foo, 9, 0] => foo, 8, 0 
[foo, 9, 1] => foo, 8, 0 
[foo, 9, 2] => foo, 8, 1 
[foo, 9, 49] => foo, 8, 48 
[foo, 9, 51] => foo, 8, 50 
[foo, 10, 0] => foo, 9, 0 
[foo, 10, 1] => foo, 9, 0 
[foo, 10, 2] => foo, 9, 1 
[foo, 10, 49] => foo, 9, 48 
[foo, 10, 51] => foo, 9, 50 
[foo, 11, 0] => foo, 10, 0 
[foo, 11, 1] => foo, 10, 0 
[foo, 11, 2] => foo, 10, 1 
[foo, 11, 49] => foo, 10, 48 
[foo, 11, 51] => foo, 10, 50 
[foo, -1, 0] => foo, -2, 0 
[foo, -1, 1] => foo, -2, 0 
[foo, -1, 2] => foo, -2, 0 
[foo, -1, 49] => foo, -2, 47 
[foo, -1, 51] => foo, -2, 49 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 2] => Aged Brie, -1, 4 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 51] => Aged Brie, -1, 51 
[Aged Brie, 9, 0] => Aged Brie, 8, 1 
[Aged Brie, 9, 1] => Aged Brie, 8, 2 
[Aged Brie, 9, 2] => Aged Brie, 8, 3 
[Aged Brie, 9, 49] => Aged Brie, 8, 50 
[Aged Brie, 9, 51] => Aged Brie, 8, 51 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 1] => Aged Brie, 9, 2 
[Aged Brie, 10, 2] => Aged Brie, 9, 3 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 10, 51] => Aged Brie, 9, 51 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 2] => Aged Brie, 10, 3 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 51] => Aged Brie, 10, 51 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 2] => Aged Brie, -2, 4 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 51] => Aged Brie, -2, 51 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 2] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 51] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 9, 0] => Backstage passes to a TAFKAL80ETC concert, 8, 2 
[Backstage passes to a TAFKAL80ETC concert, 9, 1] => Backstage passes to a TAFKAL80ETC concert, 8, 3 
[Backstage passes to a TAFKAL80ETC concert, 9, 2] => Backstage passes to a TAFKAL80ETC concert, 8, 4 
[Backstage passes to a TAFKAL80ETC concert, 9, 49] => Backstage passes to a TAFKAL80ETC concert, 8, 50 
[Backstage passes to a TAFKAL80ETC concert, 9, 51] => Backstage passes to a TAFKAL80ETC concert, 8, 51 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 9, 3 
[Backstage passes to a TAFKAL80ETC concert, 10, 2] => Backstage passes to a TAFKAL80ETC concert, 9, 4 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 51] => Backstage passes to a TAFKAL80ETC concert, 9, 51 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 2] => Backstage passes to a TAFKAL80ETC concert, 10, 3 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 51] => Backstage passes to a TAFKAL80ETC concert, 10, 51 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 2] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 51] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, 2] => Sulfuras, Hand of Ragnaros, 0, 2 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 51] => Sulfuras, Hand of Ragnaros, 0, 51 
[Sulfuras, Hand of Ragnaros, 9, 0] => Sulfuras, Hand of Ragnaros, 9, 0 
[Sulfuras, Hand of Ragnaros, 9, 1] => Sulfuras, Hand of Ragnaros, 9, 1 
[Sulfuras, Hand of Ragnaros, 9, 2] => Sulfuras, Hand of Ragnaros, 9, 2 
[Sulfuras, Hand of Ragnaros, 9, 49] => Sulfuras, Hand of Ragnaros, 9, 49 
[Sulfuras, Hand of Ragnaros, 9, 51] => Sulfuras, Hand of Ragnaros, 9, 51 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 1] => Sulfuras, Hand of Ragnaros, 10, 1 
[Sulfuras, Hand of Ragnaros, 10, 2] => Sulfuras, Hand of Ragnaros, 10, 2 
[Sulfuras, Hand of Ragnaros, 10, 49] => Sulfuras, Hand of Ragnaros, 10, 49 
[Sulfuras, Hand of Ragnaros, 10, 51] => Sulfuras, Hand of Ragnaros, 10, 51 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 1 
[Sulfuras, Hand of Ragnaros, 11, 2] => Sulfuras, Hand of Ragnaros, 11, 2 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 51] => Sulfuras, Hand of Ragnaros, 11, 51 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, 2] => Sulfuras, Hand of Ragnaros, -1, 2 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 51] => Sulfuras, Hand of Ragnaros, -1, 51 
